TodoList/TodoList.Api.Tests/Controllers/NodesControllerTest.cs

```diff
@@ -24,18 +25,52 @@ internal class NodeControllerTest
 
         public NodesController Controller;
 
+        private INodesRepository MockRepository()
+        {
+            var repository = Substitute.For<INodesRepository>();
+
+            repository.GetAllAsync().Returns(new[]
+            {
+                new NodeModel {Id = FirstId, Text = "poopy"},
+                new NodeModel {Id = SecondId, Text = "GEARS"},
+                new NodeModel {Id = ThirdId, Text = "Planet Music"},
+                new NodeModel {Id = FourthId, Text = "Time to get shwifty"}
+            });
+
+            repository.GetByIdAsync(FirstId)
+                .Returns(new NodeModel {Id = FirstId, Text = "poopy"});
+
+            repository.AddAsync(new NodeModel()).ReturnsForAnyArgs(new NodeModel
+            {
+                Id = SecondId,
+                Text = "GEARS"
+            });
+
+            repository.UpdateAsync(new NodeModel()).ReturnsForAnyArgs(new NodeModel
+            {
+                Id = ThirdId,
+                Text = "Planet Music"
+            });
+
+            return repository;
+        }
+
         [SetUp]
         public void SetUp()
         {
-            Controller = GetControllerForTests();
+            Controller = GetControllerForTests(MockRepository());
         }
 
-        private static NodesController GetControllerForTests()
+        private static NodesController GetControllerForTests(INodesRepository repository)
         {
-            return new NodesController
+            var locationHelper = Substitute.For<ILocationHelper>();
+
+            locationHelper.GetLocation(new HttpRequestMessage(), new Guid()).ReturnsForAnyArgs("api/v1/nodes/id");
+
+            return new NodesController(repository, locationHelper)
```

Comment:
this would be enough: 
```C#
           return new NodesController(repository, locationHelper)
           {
                Configuration = new HttpConfiguration(),
                Request = new HttpRequestMessage()
           };
```